edweiss-app/app/(app)/courses/[id]/index.tsx

```diff
@@ -125,39 +119,54 @@ const CoursePage: ApplicationRoute = () => {
 		});
 
 	// Filter previous assignments
-	const previousAssignments = assignments.filter((assignment) => assignment.dueDate.seconds * timeInMS.SECOND <= new Date().getTime()).reverse();
+	const previousAssignments = useMemo(() => {
+		return assignments
+			.filter(
+				(assignment) =>
+					assignment.dueDate.seconds * timeInMS.SECOND <= new Date().getTime()
+			)
+			.reverse();
+	}, [assignments, timeInMS.SECOND]);
 
 	// Filter upcoming assignments
-	const upcomingAssignments = assignments.filter((assignment) => assignment.dueDate.seconds * timeInMS.SECOND > new Date().getTime());
+	const upcomingAssignments = useMemo(() => {
+		const filteredAssignments = assignments.filter(
+			(assignment) =>
+				assignment.dueDate.seconds * timeInMS.SECOND > new Date().getTime()
+		);
+		return filteredAssignments.length > 0 ? filteredAssignments : undefined;
+	}, [assignments, timeInMS.SECOND]);
+
+	//Checks
+	if (!isValidId) { return <Redirect href={'/'} />; }
+	if (course == undefined || firebase_data == undefined) { return <TActivityIndicator size={40} />; }
 
 	return (
 		<>
 			{/* Utilisation du RouteHeader pour afficher le titre du cours */}
-			<RouteHeader title={course.data.name} align="center" isBold={true} />
+			<RouteHeader title={course.data.name} align="center" isBold />
 
 			{/* ScrollView pour permettre le défilement */}
 			<TScrollView testID={testIDs.scrollView} p={16} backgroundColor="mantle" >
 
 				{/* Section des Pending Assignments */}
-				<TText mb={10} size={18} color='darkBlue' bold={true} testID={testIDs.upcomingAssignments} >{t(`course:upcoming_assignment_title`)}</TText>
-
-				{/* Liste des assignments avec map */}
-				{upcomingAssignments.length > 0
-					?
-					upcomingAssignments.map((assignment) => (
-						<AssignmentDisplay item={assignment} index={upcomingAssignments.indexOf(assignment)} isSwipeable={true} key={assignment.name} />
-					))
-					:
-					<TText size={16} testID={testIDs.noAssignmentDue} >{t('course:no_assignment_due')}</TText>
-				}
+				<TText mb={10} size={18} color='darkBlue' bold testID={testIDs.upcomingAssignments} >{t(`course:upcoming_assignment_title`)}</TText>
+
+				<For
+					each={upcomingAssignments}
+					fallback={<TText size={16} testID={testIDs.noAssignmentDue}>{t('course:no_assignment_due')}</TText>}
+				>{(assignment, index) => (
+					<AssignmentDisplay item={assignment} index={index} isSwipeable key={assignment.name} />
+				)}
+				</For>
 
 				{/* Bouton vers les Passed Assignments */}
-				<TTouchableOpacity testID={testIDs.previousAssignmentTouchable} onPress={() => router.push({ pathname: `/courses/[id]/archive`, params: { id: course.id, extraInfo: JSON.stringify(previousAssignments) } })}>
+				<TTouchableOpacity testID={testIDs.previousAssignmentTouchable} onPress={() => router.push({ pathname: `/courses/[id]/archive`, params: { id: course.id, rawAssignments: JSON.stringify(previousAssignments) } })}>
```

Comment:
You might want to have a look at `hooks/routeParameters.ts` to typesafe this logic and make refactoring easier.

Reply:
A merge-rebase might help you, but for this specific comment it's only an improvement for your quality of life, it's an additional compiler safety to ensure that you pass the correct arguments and don't worry about parsing/stringify them.